ludwig/data/dataset_synthesizer.py

```diff
@@ -111,7 +113,12 @@ def return_none(feature):
 
 
 def assign_vocab(feature):
-    feature["idx2str"] = build_vocab(feature.get("vocab_size", 10))
+    if DECODER in feature:
```

Comment:
I recommend doing that as a follow-up task after this PR is merged, because what we have here works and the follow-up task is parallelizable AFAIK nothing else we're doing depends on it.  Created a follow-up task in linear here:  https://linear.app/predibase/issue/MLQ-54/refactor-usage-of-encoder-configdecoder-config-features